src/server/editorServices.ts

```diff
@@ -530,9 +587,29 @@ module ts.server {
                 }
             }
             this.openFileRoots = openFileRoots;
+            if (!removedProject) {
+                var openFileRootsConfigured: ScriptInfo[] = [];
+
+                for (var i = 0, len = this.openFileRootsConfigured.length; i < len; i++) {
+                    if (info === this.openFileRootsConfigured[i]) {
+                        if (info.defaultProject.deleteOpenRef() === 0) {
+                            removedProject = info.defaultProject;
+                        }
+                    }
+                    else {
+                        openFileRootsConfigured.push(this.openFileRootsConfigured[i]);
+                    }
+                }
+
+                this.openFileRootsConfigured = openFileRootsConfigured;
+            }
             if (removedProject) {
-                // remove project from inferred projects list
-                this.inferredProjects = copyListRemovingItem(removedProject, this.inferredProjects);
+                if (removedProject.isConfiguredProject()) {
+                    this.configuredProjects = copyListRemovingItem(removedProject, this.configuredProjects);
+                }
+                else {
+                    this.inferredProjects = copyListRemovingItem(removedProject, this.inferredProjects);

```

Comment:
You could use `filter` for this. Also, `copyListRemovingItem` uses `==` instead of `===`.